client/Settings/Settings.ts

```diff
@@ -11,6 +11,7 @@ export interface Settings {
         RollMonsterHp: boolean;
         AllowNegativeHP: boolean;
         AutoCheckConcentration: boolean;
+        AutoGroupInitiative: string;
```

Comment:
You can use a string literal type here to enforce that it's always a valid string:
`AutoGroupInitiative: "None" | "By Name" | "Side Initiative";`

There's a larger refactoring you could do here by encapsulating the AutoGroupInitiative options and behaviors into a class. Let me know if you're interested in taking that on, it would make the code a bit cleaner IMO and I can provide some guidance there.

Reply:
I had considered making the InitiativeGroup a class of its own, but that didn't seem to be worth while once I got a handle on what was needed. I could see making AutoGroupInitiative its own class thought. Maybe I can do that as a fast-follow PR? Any suggestions on where that class should live, i.e. folder of its own or alongside an existing class? I'd be open to hearing any other thoughts you had around it.

Reply:
I've immediately run into an error saying `Type 'string' is not assignable to type '"None" | "By Name" | "Side Initiative"'.` I see a number of open issues in the TypeScript repo about this. I think I could get around this by defining the list everywhere one of the values could be assigned, but not sure about that. Wondering instead if this would be better implemented as an Enum, and then referencing the enum values elsewhere in the code, possibly as part of the AutoGroupInitiative encapsulation. Any ideas on this?

Reply:
Yeah, this isn't a TS bug, but it can be a bit hard to troubleshoot. Try extracting a type:
`export type AutoGroupInitiativeOption = "None" | "By Name" | "Side Initiative";`
Then use that type instead of string anywhere you're getting or setting this value:
`AutoGroupInitiative: AutoGroupInitiativeOption;`

I think hold off on adding a new class for `AutoGroupInitiative`. I'll think about a more general purpose solution to these types of user options, and your implementation is great as-is.